### PR TITLE
Add tabbed view to the performance page

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This converts the timeline section header into a tabbed view in preparation for surfacing Frame Analysis and Raster Metrics data here (both features are hidden behind flags right now).

Screen Shot 2022-04-22 at 4 17 40 PM

Screen Shot 2022-04-22 at 4 18 53 PM

We were able to delete a lot of code related to the custom tabbed view we built for storing selected frames as tabs.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart [hunk 1]
```diff
@@ -165,14 +174,80 @@ class _TimelineAnalysisContainerState extends State<TimelineAnalysisContainer>
...
+  static const emptyTimelineKey = Key('Empty Timeline');
+
+  final PerformanceController controller;
+
+  final bool processing;
+
+  final double processingProgress;
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart:204]: the code in this build method is not new. It was pulled out from the above widget.

File: packages/devtools_app/lib/src/screens/performance/performance_model.dart [hunk 2]
```diff
@@ -490,6 +490,16 @@ class FlutterFrame {
   /// Timeline event data for this [FlutterFrame].
   final FrameTimelineEventData timelineEventData = FrameTimelineEventData();
 
+  FrameAnalysis? get frameAnalysis {
+    if (_frameAnalysis != null) return _frameAnalysis!;
```
Comment (elliette) [inline packages/devtools_app/lib/src/screens/performance/performance_model.dart:494]: Can we remove the `!`?

File: packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart [hunk 3]
```diff
@@ -46,117 +49,124 @@ class TimelineAnalysisContainer extends StatefulWidget {
...
+      tabViews: tabViews,
+      gaScreen: analytics_constants.performance,
+      // TODO(kenz): enable analytics when this view is stable.
+      sendAnalytics: false,
     );
   }
 
+  List<DevToolsTab> _generateTabs() {
+    final searchFieldEnabled =
+        !(controller.data?.isEmpty ?? true) && !widget.processing;
```
Comment (elliette) [inline packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart:120]: Optional: pull `!(controller.data?.isEmpty ?? true)` into a variable like `hasData` to make this easier to read

**Review by elliette — APPROVED** [review]
LGTM with a couple small suggestions!

File: packages/devtools_app/lib/src/screens/performance/performance_model.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/performance_model.dart]: done

File: packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart]: done

**Pull request merged**